### PR TITLE
fix: do not allow special characters in username

### PR DESCRIPTION
**Pull request by MatveyK**
Relates to https://github.com/graasp/graasp/pull/1406

### CODE REVIEW

File: src/langs/it.json [hunk 1]
```diff
@@ -115,4 +116,4 @@
   "NOT_FOUND_PAGE_TEXT": "Pagina non trovata",
   "NOT_FOUND_PAGE_MESSAGE": "Spiacenti, non siamo riusciti a trovare questa pagina.",
   "GO_TO_HOME_TEXT": "Vai a Home"
-}
+}
```
Comment (spaenleh) [inline src/langs/it.json:119]: Can you check that your editor is setup to add an empty new line at the end of files ?
```
  "files.trimTrailingWhitespace": true
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Thank you for this fix, can you add a test that verifies that when inputing a forbidden character the user sees the error message ?

**Comment by spaenleh** [comment]
@MatveyK Also see about the issue raised by sonarCloud. Maybe there is way to make the regex simpler ? Not sure we need backtracking in there ?

File: src/langs/it.json (hunk 1 above)
Comment (MatveyK) [reply src/langs/it.json:119]: Added.

**Comment by MatveyK** [comment]
Regex simplified and test added.

**Review by pyphilia — APPROVED** [review]
LGTM, same comment as in https://github.com/graasp/graasp-auth/pull/457#pullrequestreview-2328448199

**Review by spaenleh — APPROVED** [review]

**Pull request merged**